### DISCUSSION TITLE
How to Configure Split DNS / Horizon

### DISCUSSION THREAD

**Discussion by Flagelmann** [TechnitiumSoftware/DnsServer#815] [Q&A] [answered]
Hello,

I found out only other two entries about this topic, Split Horizon / DNS, but unfortunately I did not get completely the info that I need in a clear way.

As it would be very useful to use this app but there's lack of any detailed, scenario-based, documentation of this app, I would like to ask if any of you has finally configured it properly on his own self-hosted instance.

The scenario in which I am is the following:

- I have a primary DNS zone configured and working correctly, with several A records registered.
- This DNS server is serving both my LAN but also some personal devices from internet, mainly using DoT configuration (Private DNS).
- The DNS domain used is and MUST remain the same for both internal and external requests but providing a different IP based on where the device is located (and here it is the need of Split DNS).

Now, I have checked other few posts I have found about the Split Horizon app and it seems it is done by doing two main steps:

1. Opening the "Config" part from the app page and modifying/saving the main configuration file of the app (where specifying the networkGroupMap and the groups basically).
2. Then creating an "APP" record in which we have to define in the "Record Data" some other spec about the actual translation needed.

NOTE: need also to understand better the "externalToInternalTranslation" purpose defined under each group.

Personally, I get the idea of the configuration, but still do not get a full picture of the configuration in Technitium DNS server, as there is a complete lack of a full example, so the "main configuration" + "APP record" properly configured and explained.

Right now to configure it for me it seems basically a trial and error.

Anyone that could provide a full example on how to properly configure the Split Horizon app for this scenario (no need of a CNAME example, but A record example)?

Many thanks

**Comment by ShreyasZare** [comment]
Thanks for asking. Unfortunately there is no documentation available for the apps. However, there is plan to support GUI for all apps so it should be more user friendly and easy to understand once GUI is available.

The Split Horizon app supports 2 features. First is to use APP record which is useful only when you have the zones locally hosted. With this APP record, you can return custom A or CNAME response based on the request's IP address.

The second feature is Address Translation which works globally (for all domain names hosted or not) and is to be configured via the app's config. In there, you need to configure groups for the client networks and in that group you can specify the external to internal translation map. This feature will automatically check for all the responses and replace any external IP match with internal IP as defined in the config.

Let me know if you have any more queries regarding the config.

**Comment by Flagelmann** [comment]
Thanks for your availability ShreyasZare. :)

Ok, let's say that I want to use the APP record approach and not the global config approach.

I go to the actual zone, then I open the new APP record section, as shown below.

Then I want to specify, for the specific subdomain, let's say **g.example.com** that for internal LAN based on 192.168.1.0/25 the IP that should be returned to clients will be 192.168.1.40 and for all the others the public IP one, how should I configure the "Record Data" section?

![image](https://github.com/TechnitiumSoftware/DnsServer/assets/72081421/30e02db6-0410-45df-813a-b8bc44fc6307)

**Reply by ShreyasZare** [reply, marked as answer]
You're welcome :)

The record data JSON for that scenario will be as follows:
```
{
  "192.168.1.0/25": [
    "192.168.1.40"
  ],
  "0.0.0.0/0": [
    "<public-ip-address>"
  ]
}
```

The `0.0.0.0/0` network address will match to all addresses. 

Edit: fixed missing comma in example.

**Reply by Flagelmann** [reply]
Perfect. :D

Ok, this is a first common simple scenario that would be very helpful to introduce to Split DNS in Technitium.

Thanks a lot, by doing some first tests even using Private DNS on multiple Android devices it seems working as expected, getting the proper IP as expected.

There is a missing comma in the example above, but fixed as follows:

```
{
  "192.168.1.0/25": [
    "192.168.1.40"
  ],
  "0.0.0.0/0": [
    "<public-ip-address>"
  ]
}
```

**Reply by ShreyasZare** [reply]
Good to know you got it working. Will edit to fix missing comma.

**Reply by UncleVic** [reply]
I have the same question because I'd just faced out with Split a week ago.

What about A records? Do I have to remove them completely or not?

**Reply by ShreyasZare** [reply]
> I have the same question because I'd just faced out with Split a week ago.
> 
> What about A records? Do I have to remove them completely or not?

If you have A records then those will get returned. So you will need to remove them so that the requests hits APP record.

**Reply by UncleVic** [reply]
As far as I understand, I can't use DNSSEC with Split. Right?

**Reply by ShreyasZare** [reply]
Ya, the current DNSSEC support signs the zone records in advance so the records cannot change dynamically. Online signing would allow such dynamic records and its planned but will take a while as its not being prioritized.

For now, there is another option where you can sign your zone and create another zone for the subdomain name such that this sub domain zone is unsigned. So, you will be able to sign all the required records and have only the dynamic APP records separately working in the sub domain zone.

**Reply by UncleVic** [reply]
I set up my server by your advice, but when I try making a request from public to my server, I've got nothing
The A records can resolve, but APP record can't.

It's possible there is a problem in the main split config. It's here:
```json
{
    "networks": {
        "custom-networks": []
    },
    "enableAddressTranslation": true,
    "networkGroupMap": {
        "192.168.1.0/24": "local",
        "172.16.0.0/16": "docker"
    },
    "groups": [
        {
            "name": "local",
            "enabled": true,
            "translateReverseLookups": true,
            "externalToInternalTranslation": {}
        },
        {
            "name": "docker",
            "enabled": true,
            "translateReverseLookups": true,
            "externalToInternalTranslation": {}
        }
    ]
}
```
How does main config affect APP records?

**Reply by ShreyasZare** [reply]
Thanks for asking. The Split Horizon's main config is for Address Translation feature and APP records only use the "networks" part where you can define a named network group to use.

If you have A record and APP record for the same domain name then the APP record will not be used.

**Reply by UncleVic** [reply]
Maybe you didn't understand me.
If I create A record (ex. ns1.domain.info), I can resolve it from public networks.
if I replace this record by APP once 
```json
{
  "192.168.1.0/24": [
    "192.168.1.2"
  ],
  "0.0.0.0/0": [
    "my public IP"
  ]
}
```
I can't resolve it from public, but I can resolve it from my local network.

**Reply by UncleVic** [reply]
And some more...
If I replace APP record by wrong network ex.
```json
{
  "192.168.222.0/24": [
    "192.168.1.2"
  ],
  "0.0.0.0/0": [
    "my public IP"
  ]
}
```
it'll be resolved from my local network as my public IP. It's clear...
But why can't I resolve it from public?

**Reply by ShreyasZare** [reply]
Thanks for the config. I tested this exact config and its working in my test setup. 

Enable Query Logging option and check the source IP for the requests that come from when you test it from public Internet. I guess there is some issue with the source IP for the received request.

**Reply by UncleVic** [reply]
Thank for your patience! It seems I understood!
When I requested my domain, I've got the answer from my provider's server.
When I tried to request particular my server, I've got a correct answer.

Now I should solve this problem with my provider.

Thanks a lot!

**Reply by ShreyasZare** [reply]
You're welcome!

**Reply by UncleVic** [reply]
Sorry for I'm asking again, but I couldn't solve this problem with my provider.

Let's start with my main config
```json
{
    "networks": {
        "custom-networks": []
    },
    "enableAddressTranslation": true,
    "networkGroupMap": {
        "192.168.1.0/24": "local",
        "172.16.0.0/16": "docker" 
    },
    "groups": [
        {
            "name": "local",
            "enabled": true,
            "translateReverseLookups": true,
            "externalToInternalTranslation": {}
        },
        {
            "name": "docker",
            "enabled": true,
            "translateReverseLookups": true,
            "externalToInternalTranslation": {}
        }
    ]
}
```
Here, `"172.16.0.0/16": "docker" ` just for the other group map

Ok... I have NS and A records of my NS-server
```
@                     3600      IN  NS            ns1
ns1                   3600      IN  A             109.173.40.133
```
And I have a simple APP record for test
```
fs                    3600      IN  APP           "Split Horizon" "SplitHorizon.SimpleAddress" 
```
```json
{
  "192.168.1.0/24": [
    "192.168.1.3"
  ],
  "0.0.0.0/0": [
    "109.173.40.133"
  ]
}
``` 

If I make a request from my local network
```sh
❯ nslookup fs.domain.com
Server:		192.168.1.2
Address:	192.168.1.2#53

Name:	fs.domain.com
Address: 192.168.1.3

# the next one
❯ nslookup fs.domain.com ns1.domain.com
Server:		ns1.domain.com
Address:	109.173.40.133#53

Non-authoritative answer:
*** Can't find fs.domain.com: No answer
```
If I make a request from my external server
```sh
nslookup fs.domain.com ns1.domain.com
Server:		ns1.domain.com
Address:	109.173.40.133#53

Name:	fs.domain.com
Address: 109.173.40.133

# the next one
nslookup fs.domain.com
Server:		1.1.1.1
Address:	1.1.1.1#53

Non-authoritative answer:
*** Can't find fs.domain.com: No answer
```

Moreover, I have the Secondary DNS server is hosted by my provider.
And I can see in my logs that the Secondary DNS gets my zone without any errors, but...
```sh
nslookup fs.domain.com ns8-l2.nic.com
Server:		ns8-l2.nic.com
Address:	91.217.21.20#53

*** Can't find fs.domain.com: No answer
```

What is wrong?
My provider says that he understands Split, but I can't request my zone from my Secondary DNS.
I tried to create Secondary DNS locally and it works fine.
And my main problem is that my Secondary DNS is hosted by my provider doesn't understand my zone. :(

**Reply by ShreyasZare** [reply]
@UncleVic thanks for the details. The APP records are proprietary and will only work with Technitium DNS server which is why your secondary DNS provides is failing to return an answer for the request. If you need to have this setup then you will need to host your own secondary DNS servers. 

The other option is to delegate a subdomain name as a separate zone and host it independently on Technitium DNS server while rest of you zone is hosted with your existing secondary DNS provider. On this subdomain zone, you can then use the proprietary features.

**Reply by UncleVic** [reply]
Yeah... Unfortunately, I see... I understood it yesterday after running the other Technitium as Secondary

Sorry again :(

**Comment by kranked** [comment]
Sorry to revive an old thread, but can you point it to a ANAME? e.g.
{
  "192.168.1.11": [
    "<public-ip-address>"
  ],
  "0.0.0.0/0": [
    "webserverexample.lan"
  ]
}

**Reply by ShreyasZare** [reply]
You can do that using the `SplitHorizon.SimpleCNAME` class path option.

**Reply by kranked** [reply]
ah that should of been obvious, thank you!

**Reply by ShreyasZare** [reply]
You're welcome!

**Reply by kranked** [reply]
I have this app record set for tplinkcloud.com however the 0.0.0.0/0 range is still receiving the public IP, any ideas? (also tried the subnet the DHCP serves specifically, 192.168.7.0/24 as in screenshot.) 

![image](https://github.com/user-attachments/assets/939c208f-0ffa-4159-b640-312ed51efa2c)

I also get the following message if I set up the same record for the specific subnet the camera is trying to resolve. - n-device-api.tplinkcloud.com

<html><body>

2024-11-13 19:37:34 | 192.168.7.106 | Udp | Authoritative | ServerFailure | n-device-api.tplinkcloud.com
-- | -- | -- | -- | -- | --



</body>
</html>

Extended error from DNS client:

{
  "Metadata": {
    "NameServer": "tuvipdnsserver1 (192.168.3.62)",
    "Protocol": "Udp",
    "DatagramSize": "57 bytes",
    "RoundTripTime": "3.01 ms"
  },
  "EDNS": {
    "UdpPayloadSize": 1232,
    "ExtendedRCODE": "ServerFailure",
    "Version": 0,
    "Flags": "None",
    "Options": []
  },
  "DnsClientExtendedErrors": [
    {
      "InfoCode": "NetworkError",
      "ExtraText": "tuvidnsserver1 (192.168.3.62) returned RCODE=ServerFailure for n-device-api.tplinkcloud.com. A IN"
    }
  ],
  "Identifier": 63900,
  "IsResponse": true,
  "OPCODE": "StandardQuery",
  "AuthoritativeAnswer": false,
  "Truncation": false,
  "RecursionDesired": true,
  "RecursionAvailable": true,
  "Z": 0,
  "AuthenticData": false,
  "CheckingDisabled": false,
  "RCODE": "ServerFailure",
  "QDCOUNT": 1,
  "ANCOUNT": 0,
  "NSCOUNT": 0,
  "ARCOUNT": 1,
  "Question": [
    {
      "Name": "n-device-api.tplinkcloud.com",
      "Type": "A",
      "Class": "IN"
    }
  ],
  "Answer": [],
  "Authority": [],
  "Additional": [
    {
      "Name": "",
      "Type": "OPT",
      "Class": "1232",
      "TTL": "0 (0 sec)",
      "RDLENGTH": "0 bytes",
      "RDATA": {
        "Options": []
      },
      "DnssecStatus": "Disabled"
    }
  ]
}

**Reply by ShreyasZare** [reply]
Thanks for the details. Since you are using SimpleCNAME class which generates a CNAME record, you cannot use an IP address in the config. You will need to use a domain name and then setup an A record for that domain name to resolve the request correctly.

**Reply by kevdogg** [reply]
I'm using the APP feature of split horizon.  With your example:

```
{
  "192.168.1.0/25": [
    "192.168.1.40"
  ],
  "0.0.0.0/0": [
    "<public-ip-address>"
  ]
}
```
Say if for the first network, you actually wanted to define multiple networks, for example:

```
{
  "192.168.1.0/25", "192.168.2.0/24" : [
    "192.168.1.40"
  ],
  "0.0.0.0/0": [
    "<public-ip-address>"
  ]
}
```
How would you do this?  I have a feeling you would create a network group and then reference the group name here, however I'm just missing some documentation here.

**Reply by ShreyasZare** [reply]
@kevdogg thanks for asking. As you guessed, you need to configure all the networks in the app's main config under "networks" property with a name. Then use the configured name with your APP record.

**Reply by kevdogg** [reply]
Hey you don't have the format for the json documented anywhere?  I know it was there but once I started typing the format went away and I can't seem to make it reappear.  Not saying you documentation but maybe the format could be listed somewhere for an example.

**Reply by ShreyasZare** [reply]
@kevdogg Thanks for asking. The json template is available in the Apps section under the "More Details" link for the Split Horizon app. Just click that link to expand the details.

**Reply by UncleVic** [reply]
Sorry for asking again, but it doesn't work.

My Split's main config is
```
{
    "networks": {
        "public": [
          "0.0.0.0/0"
        ],
        "private": [
          "192.168.10.0/24",
          "192.168.20.0/24"
        ]
    }
}
```

The DNS record is
```
{
  "private": [
    "192.168.20.2"
  ],
  "0.0.0.0/0": [
    "10.17.40.1"
  ]
}
```

And the name is resolved with the external IP.

**Reply by ShreyasZare** [reply]
@UncleVic thanks for asking. You cannot define "public" and "private" networks since those are keywords so you need to use different names for your networks in the app's main config.

**Reply by UncleVic** [reply]
Are there any more keywords?

**Reply by ShreyasZare** [reply]
@UncleVic No. Just public and private are the ones used to automatically map to all public/private networks.

**Comment by smileBeda** [comment]
Sorry to revive the older thread, I would like to achieve something similar but more globally.

I’d like to achieve a global Split Horizon behavior for my DNS zones:
- If a request comes from the `10.7.0.0/24` subnet (e.g., from `10.7.0.4`), I want all my internal services registered in Technitium (e.g., `heimdall.lan`, `jellyfin.lan`, etc.) to resolve to their WireGuard IP (`10.7.0.3`).
- If a request comes from the `192.168.1.0/24` subnet (e.g., from `192.168.1.4`), I want the same services to resolve to their LAN IP (`192.168.1.3`).
- For all other domains (e.g., external sites like google.com), I want Technitium to behave like a normal resolver with no translation - just forward the query to the upstream DNS.

I understand that to achieve this I have to:
- install the Split Horizon Plugin
- directly edit the config - not the single records
- however, I got confused on the `dnsApp.config` syntax. Do I have to _remove_ the entries already present such as `custom-networks` and the already present groups? Perhaps an example based on above would help me greatly!
- and, of course, finally, I would have to go and remove all the existing A records, as I understand it from this thread... I guess I will have some work ahead of me haha

Thanks for any hints and feedbacks, I appreciate both answer and that tool!

**Reply by ShreyasZare** [reply]
Thanks for asking. You do not need to edit the dnsApp.config, especially not to edit the config file directly since the changes wont apply unless you save it from the GUI.

To make it work, just add an APP record in your conditional forwarder zone and then configure the record's json config. The json is structured in such that you use the client network as the key and enter the addresses to return as an array to it. If you do not specify the `default` key, they APP record wont have any output and the request will get forwarded based on the FWD record in the forwarder zone.

**Reply by smileBeda** [reply]
I cannot get this to work

1. If I set the rule as a forwarder zone, it simply never "overwrites" single entries (which still reply with the set A record of that entry, even if I delete the A record of that entry)
2. If I set the rules for each entry, then yes I get proper reply (A record either 10.xxx or 192.xxx) however, then browsers do not load the domains saying is not using SSL, and if I accept the risk it says "cannot find server". When I `dig` the domain, I can see it returns the right A records and uses the right DNS, yet, `curl` will fail, unless I `curl -v --resolve heimdall.lan:443:10.7.0.4 https://heimdall.lan` directly bypassing DNS. But that is not how browsers load the stuff.


I think the problem on my end is of course mixed (one part is my connections not using the right DNS, apparently) but the other end seems to be something I am doing wrong in Technitium :(

**Reply by ShreyasZare** [reply]
Thanks for the info. If a record exists and matches the exact query then its always used.  So, you have to remove all A/AAAA records for the domain name to make it work. Only keep the APP record and the FWD record you have.

The APP record will resolve the domain name as you have configured and that ends the DNS role here. The browser SSL warning will come since you may be running a self signed cert on your local web server. If you own the domain name then you can copy the cert from your public web server to your local web server to make it work without browser SSL errors. If the domain is not yours then there is nothing that can be done regarding the ssl error.

Also make sure that your browser does not have private/encrypted DNS enabled which will bypass your local DNS server. Check the browser Options and make sure that DoH or encrypted DNS is off.

**Reply by smileBeda** [reply]
The global Forwarder simply does not work for me even when removing the single entries A/AAAA

However what I got working is setting an APP record for _each_ entry (and remove A/AAAA)

JSOn in both cases is:
```
{
  "10.7.0.0/24": ["10.7.0.4"],
  "192.168.1.0/24": ["192.168.1.3"],
  "default": []
}
```

This works on the single record level. In the Zone for `.` on Forwarder this does not work and presents the issues aforementioned (I am aware on the SSL issue, but that is not the problem because I already installed the cert, otherwise it would not work either when not using the horizons feature)

Perhaps this simply is not how global forwarder is working, and I misuderstand it?

This is how I set it up:
<img width="762" alt="Screenshot 2025-06-03 at 11 26 17" src="https://github.com/user-attachments/assets/dcb0ef31-e7a7-4733-a9a3-8d022a48ef0b" />
<img width="774" alt="Screenshot 2025-06-03 at 11 26 28" src="https://github.com/user-attachments/assets/85c74ac2-5688-404f-b62a-835cadb67b3a" />

And then in the zone I add `@` record with the aforeshared JSON

**Reply by ShreyasZare** [reply]
Thanks for the details. Make the APP record's json config as below:
```
{
  "10.7.0.0/24": ["10.7.0.4"],
  "192.168.1.0/24": ["192.168.1.3"]
}
```

The empty `default` entry days that there is no A/AAAA records so a NO DATA response is sent. When that entry is missing, the APP record bails and let the forwarding work.

Also, make the APP record a wildcard by using `*` as the subdomain name if you need that all domain names resolve to that IP. Otherwise, have a single APP records with a specific different subdomain name and then using a CNAME that points to the APP record for all the other domain names you wish to resolve. This way would be easier than having APP record for all of them.

**Reply by smileBeda** [reply]
Thanks, I will try this in short.
> Also, make the APP record a wildcard by using * as the subdomain name if you need that all domain names resolve to that IP.

Do you mean where I added `.` in the `Zone`, I add `*` instead?
Note, I have no subdomains - only top level domains, such as `heimdall.lan` or `jellyfin.lan`
All non-lan is of course to be routed over standard DNS like 1.1.1.1 or else (bypass technitum basically)

UPDATE:
no, that cannot be it, because if I add `*` I get `Error! Domain name for a zone cannot contain wildcard character.` when setting up the Forwarder zone.

**Reply by ShreyasZare** [reply]
> Do you mean where I added . in the Zone, I add * instead?

I meant to use `*` when adding the APP record. You enter the `*` in the Name field of the APP record.

>  Note, I have no subdomains - only top level domains, such as heimdall.lan or jellyfin.lan
> All non-lan is of course to be routed over standard DNS like 1.1.1.1 or else (bypass technitum basically)

If you wish to manage only `lan` and a few other domain names, then I would recommend that you create forwarder zone for `lan` and the other ones separately instead of having a single root forwarder zone. This will make it easier to manage and debug issues.

**Reply by smileBeda** [reply]
It works flawlessly as long I create an entry in _each zone_. If I create a root zone, and add an App record, it does not work.
Then the error happens I mentioned before (first, says cert is invalid, and if accepted, says server not found).
Yet it works perfectly if I add it to _each single zone_.

**Comment by raylu** [comment]
is there a way to get split horizon to handle IPv6 addresses? I tried
```
{
  "127.0.0.0/8": ["127.0.0.1"],
  "::1": ["127.0.0.1"],
  "::1/128": ["127.0.0.1"],
  "0.0.0.0/0": ["127.2.3.4"]
}
```
and when I query technitium from the machine it's running on, I get
```
$ dig xis
[...]
;; ANSWER SECTION:
xis.                    3600    IN      A       127.2.3.4

;; Query time: 0 msec
;; SERVER: ::1#53(::1) (UDP)
```

**Reply by ShreyasZare** [reply]
Check the query logs and find out what was the source IP address for the dig request that you made and verify if it matches the APP record config. The app will respond based on the source IP address that was used.

Another thing, `::1` and `::1/128` are one and the same thing so just remove one of them from the config.

You can enable query logging from the Settings > Logs section which will log all queries to the text log file. Or, you can install Query Logs (Sqlite) app and then use the Logs > Query Logs section.

**Reply by raylu** [reply]
ah, thanks a lot! I was able to pull the logs from the regular Logs → View Logs section. technitium is running in docker so the client IP when run on the host is 172.17.0.0/16

(I know `::1` and `::1/128` are the same but neither was working so I kept trying stuff)

**Reply by ShreyasZare** [reply]
You're welcome! Docker networking is quirky.

**Comment by ttsak** [comment]
Hello

I see in Split Horizon app version 10.0 that the default config is this:

```
{
    "appPreference": 40,
    "networks": {
        "custom-networks": [
            "172.16.1.0/24",
            "172.16.10.0/24",
            "172.16.2.1"
        ]
    },
    "enableAddressTranslation": false,
    "domainGroupMap": {
        "example.com": "local1"
    },
    "networkGroupMap": {
        "10.0.0.0/8": "local1",
        "172.16.0.0/12": "local2",
        "192.168.0.0/16": "local3"
    },
    "groups": [
        {
            "name": "local1",
            "enabled": true,
            "translateReverseLookups": true,
            "externalToInternalTranslation": {
               "1.2.3.0/24": "10.0.0.0/24",
               "5.6.7.8": "10.0.0.5"
            }
        },
        {
            "name": "local2",
            "enabled": true,
            "translateReverseLookups": true,
            "externalToInternalTranslation": {
               "1.2.3.4": "172.16.0.4",
               "5.6.7.8": "172.16.0.5"
            }
        },
        {
            "name": "local3",
            "enabled": true,
            "translateReverseLookups": true,
            "externalToInternalTranslation": {
               "1.2.3.4": "192.168.0.4",
               "5.6.7.8": "192.168.0.5"
            }
        }
    ]
}
```

First of all , as I understand:

* The definitions in `network` section are only used for the proprietary `APP` records in the zones
* Everything else except the `networks` section are only used when `enableAddressTranslation` is set to `true`

Are these assumptions correct ?

And second , can you elaborate a bit on the way that `networkGroupMap` and `domainGroupMap` sections are used?
The networkGroupMap definitions add client ip addresses unnder group names , which are then used below for the translations.
But what is domainGroupMap used for and how does it work (what does the provided example do)?

**Reply by ShreyasZare** [reply]
Thanks for asking.

> The definitions in network section are only used for the proprietary APP records in the zones

Yes, its only used with the APP records.

> Everything else except the networks section are only used when enableAddressTranslation is set to true

Ya, rest of the config is related to the address translation feature.

> And second , can you elaborate a bit on the way that networkGroupMap and domainGroupMap sections are used?

The `networkGroupMap` entries will map a client to a group so that the address translation settings in the group are used. The `domainGroupMap` entries map a domain name to a group for same reason but this allows you to have different domain names translated using different rules. The `domainGroupMap` map entries have higher priority than the `networkGroupMap` and you should use them depending on what your scenario demands.
